### PR TITLE
[FIX] project: review UX and UI in Project app

### PR DESCRIPTION
**Pull request by RaoufGhrissi**
The following changes are made in this commit:

PAKY:
- swap of user_id icon in the kanban box and the removal of the allow_subtasks field and setting
- change the ordering of the fields in the subtask list
- make small layout changes in the project.task and project.project kanban cards
- remove the "lock-icon private" sub-title of private tasks, replace it with a little lock icon on the bottom right icons of the kanban card.
- removed the break tag in the project.task kanban card that was unecessary given the new display and margin style settings
- lock icon is bigger
- kanban icons are better aligned
- state is at the right of subtask list
- state is as big as avatar

BVDN:
- project.task kanban card > the state should have the same size as the avatar
- sub-tasks list view
- remove the 'remaining hours on SO' field
- the deadline should be optional but hidden by default
- remove the rating field
- add stage option
- apply the same changes to the blocked by list view
- the priority and the state should not be optional
- project.task kanban view > change the opacity of the tasks that are closed
- the stage_id should be copied when duplicating a task, except when the task is generated through the recurrence
- Field Service task > Mark as Done > should set the 'status' to 'done' and shouldn't write on the stagec
- remove the tooltip of the tag_ids field
- the striked should be replaced by the opacity on the kanban card

ABGH:
- When duplicating a task having sub-tasks, '(copy)' is not added to the name of the sub-tasks of this task.
- project.task kanban view:
* (+ x tasks) mention next to the name is removed
* The caret is replaced with 'fa-check-square-o x/y' which should represent the number of sub-tasks open compared to the total number of sub-tasks
* Only open subtasks are displayed
* When changing the state of a sub-task to a closing one, the sub-task is muted and removed from the list on the view reload
* The name of the parent task on the kanban card of sub-tasks is displayed except when viewing the sub-tasks of a particular task through the sub-tasks stat button
* project.project kanban view: the fa-check-square-o icon of milestones is replaced with fa-flag-o
* project.task kanban card: the fa-play and fa-pause icons are moved on the right of the remaining hours widge.
* Allow users to edit the stage_id in batch from the list view of tasks if all of the selected tasks are part of the same project.

Related PRs:
(16.2)
Enterprise: https://github.com/odoo/enterprise/pull/38346

(Master - deprecated)
Enterprise: odoo/enterprise#38132

Task-3229873

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/addons/base/models/res_config.py [hunk 1]
```diff
@@ -346,16 +346,16 @@ class MyConfigWizard(models.TransientModel):
             The attribute 'group' may contain several xml ids, separated by commas.
 
         *   For a selection field like 'group_XXX' composed of 2 string values ('0' and '1'),
-            ``execute`` adds/removes 'implied_group' to/from the implied groups of 'group', 
+            ``execute`` adds/removes 'implied_group' to/from the implied groups of 'group',
```
Comment (mart-e) [inline odoo/addons/base/models/res_config.py:349]: Could you disable the auto-linter of your editor to avoid useless diff?

File: addons/project/views/project_views.xml [hunk 2]
```diff
@@ -1366,19 +1357,25 @@
...
                                         <b t-if="record.rating_active.raw_value and record.rating_count.raw_value &gt; 0" groups="project.group_project_rating">
-                                            <span style="font-weight:bold;" class="fa fa-fw mt4 fa-smile-o text-success" t-if="record.rating_avg.raw_value &gt;= 3.66" title="Average Rating: Satisfied" role="img" aria-label="Happy face"/>
-                                            <span style="font-weight:bold;" class="fa fa-fw mt4 fa-meh-o text-warning" t-elif="record.rating_avg.raw_value &gt;= 2.33" title="Average Rating: Okay" role="img" aria-label="Neutral face"/>
-                                            <span style="font-weight:bold;" class="fa fa-fw mt4 fa-frown-o text-danger" t-else="" title="Average Rating: Dissatisfied" role="img" aria-label="Sad face"/>
+                                            <span style="font-weight:bold; padding-top:1.5px; padding-right:3px; font-size:16px;" class="fa fa-fw fa-smile-o text-success" t-if="record.rating_avg.raw_value &gt;= 3.66" title="Average Rating: Satisfied" role="img" aria-label="Happy face"/>
```
Comment (auon-odoo) [inline addons/project/views/project_views.xml:1364]: Such a pity that we have to write that much style. Anyway, there is no other way.
However, we could create a class `rating_face` or something, and use a style sheet to set the the style on all 3 at once.

File: addons/project/views/project_views.xml [hunk 3]
```diff
@@ -1366,19 +1357,25 @@
...
+                                            <span style="font-weight:bold; padding-top:1.5px; padding-right:3px; font-size:16px;" class="fa fa-fw fa-meh-o text-warning" t-elif="record.rating_avg.raw_value &gt;= 2.33" title="Average Rating: Okay" role="img" aria-label="Neutral face"/>
+                                            <span style="font-weight:bold; padding-top:1.5px; padding-right:3px; font-size:16px;" class="fa fa-fw fa-frown-o text-danger" t-else="" title="Average Rating: Dissatisfied" role="img" aria-label="Sad face"/>
                                         </b>
-                                        <a t-if="record.subtask_count.raw_value and record.allow_subtasks.raw_value" class="subtask_list_button" title="See Subtasks" style="padding: 0px 5px 0px 0px; box-sizing: border-box; font-size: 20px;"/>
+                                        <a t-if="!record.project_id.raw_value" class="text-muted" style="font-size: 17px; padding-top: 1.5px;"><i title="Private Task" class="fa fa-lock"></i></a>
```
Comment (auon-odoo) [inline addons/project/views/project_views.xml:1368]:
```suggestion
                                        <a t-if="!record.project_id.raw_value" class="text-muted" style="font-size: 17px; padding-top: 1.5px;">
                                            <i title="Private Task" class="fa fa-lock"/>
                                        </a>
```
- New lines
- <i> tag can be directly closed, can't it?

**Review by auon-odoo — CHANGES_REQUESTED** [review]
Thank you guys :fire: 
Huge style sheet tho. I don't like it, but it works and I have no idea how to deal with it differently. :shrug:

File: addons/project/views/project_views.xml (hunk 3 above)
Comment (pkyriakou) [reply addons/project/views/project_views.xml:1368]: I'll work on this

File: addons/project/views/project_views.xml (hunk 2 above)
Comment (pkyriakou) [reply addons/project/views/project_views.xml:1364]: I will work on this

File: addons/project/views/project_views.xml [hunk 4]
```diff
@@ -1339,24 +1327,27 @@
...
-                                        <span t-if="record.allow_milestones.raw_value and record.milestone_id.raw_value" t-attf-class="{{record.has_late_and_unreached_milestone.raw_value ? 'text-danger' : ''}}">
-                                            <br/>
-                                            <field name="milestone_id" options="{'no_open': True}" />
+                                        <span invisible="context.get('default_parent_id', False)">
+                                            <t t-if="record.parent_id.raw_value">
+                                                <br/><field name="parent_id"/>
+                                            </t>
                                         </span>
-                                        <br />
+                                        <span invisible="context.get('default_project_id', False)"><br/><field name="project_id" widget="project_private_task" options="{'no_open': True}"/></span>
```
[the remark this answers is not in the document]
Comment (pkyriakou) [reply addons/project/views/project_views.xml:1341]: I will work on this

File: addons/project/models/project.py [hunk 5]
```diff
@@ -1688,12 +1689,15 @@ def copy(self, default=None):
         if self.allow_task_dependencies and 'task_mapping' not in self.env.context:
             self = self.with_context(task_mapping=dict())
         has_default_name = bool(default.get('name', ''))
+        if self.personal_stage_id:
+            default['personal_stage_id'] = self.personal_stage_id.id
+        if self.personal_stage_type_id:
+            default['personal_stage_type_id'] = self.personal_stage_type_id.id
```
[the remark this answers is not in the document]
Comment (bastvdn) [reply addons/project/models/project.py]: Done

File: addons/project/static/src/components/project_task_state_selection/project_task_state_selection.xml [hunk 6]
```diff
@@ -7,16 +7,16 @@
...
                 </button>
             </t>
         </xpath>
         <!-- The toggle mark as done button -->
         <xpath expr="//t[@t-if='isReadonly']" position="after">
             <t t-elif="isToggleMode and currentValue == '01_in_progress'">
-                <button t-if="['kanban', 'list'].includes(this.props.viewType) or this.env.isSmall" class="d-flex align-items-center btn fw-normal p-0" t-on-click.stop="toggleState">
-                    <i t-attf-class="{{ stateIcon(currentValue) }} {{ statusColor(currentValue) }}"></i>
+                <button t-if="['kanban', 'list'].includes(this.props.viewType) or this.env.isSmall" class="d-flex align-items-center btn fw-normal p-0" t-on-click.stop="toggleState" t-att-class="{'o_task_state_list_view': this.props.viewType === 'list'}">
+                    <i t-attf-class="{{ stateIcon(currentValue) }} {{ statusColor(currentValue) }} #{['1_done','1_canceled'].includes(currentValue) and this.props.viewType === 'kanban' ? 'opacity-50' : ''}"></i>
```
[the remark this answers is not in the document]
Comment (bastvdn) [reply addons/project/static/src/components/project_task_state_selection/project_task_state_selection.xml]: We are aiming to remove is_closed field in the not so distant future, so we're trying to use it as little as possible.
Could you explain the difference between the two ways of writing the t-attf-class attribute please ? :)

[the remark this answers is not in the document]
Comment (RaoufGhrissi) [reply addons/project/static/src/components/project_task_state_selection/project_task_state_selection.xml]: @bastvdn 
There are two equivalent syntaxes for format strings: "plain_text {{code}}" (aka jinja-style) and "plain_text #{code}" (aka ruby-style).
as i saw on the internet, they recommend using jinja-style

File: addons/project/static/src/components/subtask_kanban_list/subtask_kanban_list.js [hunk 7]
```diff
@@ -12,6 +12,17 @@ import { standardWidgetProps } from "@web/views/widgets/standard_widget_props";
...
+        this.orm = useService("orm");
+        this.subTaskIds = [];
+
+        onWillStart(this.onWillStart);
+    }
+
+    async onWillStart() {
+        this.subTaskIds = await this.orm.search(this.props.record.resModel, [
+            ["parent_id", "=", this.props.record.resId],
+            ["state", "not in", ["1_done", "1_canceled"]],
```
[the remark this answers is not in the document]
Comment (RaoufGhrissi) [reply addons/project/static/src/components/subtask_kanban_list/subtask_kanban_list.js:24]: as bastien said, we are aiming to delete the field so trying to not use it

File: addons/project/static/src/views/project_task_list/project_task_list_renderer.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+/** @odoo-module */
+
+import { ListRenderer } from "@web/views/list/list_renderer";
+import { getRawValue } from "@web/views/kanban/kanban_record";
```
[the remark this answers is not in the document]
Comment (RaoufGhrissi) [reply addons/project/static/src/views/project_task_list/project_task_list_renderer.js:4]: yes weird but the method is cohrent with the list context as we want to access a cell but maybe its definition in kanban_record is weird, i dunno

[the remark this answers is not in the document]
Comment (auon-odoo) [reply addons/project/static/src/views/project_task_list/project_task_list_renderer.js:4]: ok :) We will maybe move it in a "utils" file later, but keep it like this for now :ok_hand:

File: addons/project/static/src/components/project_task_name_with_subtask_count_char_field/project_task_name_with_subtask_count_char_field.js [hunk 9]
```diff
@@ -6,7 +6,7 @@ import { formatChar } from '@web/views/fields/formatters';
 
 class ProjectTaskNameWithSubtaskCountCharField extends CharField {
     get formattedSubtaskCount() {
-        return formatChar(this.props.record.data.allow_subtasks && this.props.record.data.child_text || '');
+        return formatChar(this.props.record.data.subtask_count ? `(${this.props.record.data.closed_subtask_count}/${this.props.record.data.subtask_count} subtask${this.props.record.data.subtask_count>1 ? 's' : ''})`  : '');
```
Comment (auon-odoo) [inline addons/project/static/src/components/project_task_name_with_subtask_count_char_field/project_task_name_with_subtask_count_char_field.js:9]:
```suggestion
        return this.props.record.data.subtask_count
             ? `(${this.props.record.data.closed_subtask_count}/${this.props.record.data.subtask_count} subtasks)`
             : "";
```
Sorry I hadn't seen that before :see_no_evil: 
- Why use `formatChar`?
- Break the line.
- We don't handle singular/plural. (Think about other languages :nauseated_face: )
- Use "" i/o ''. (don't ask why, idk)

File: addons/project/static/src/components/project_task_state_selection/project_task_state_selection.xml [hunk 10]
```diff
@@ -7,19 +7,19 @@
...
                     <i class="fa fa-lg fa-hourglass-o"></i>
                 </button>
             </t>
         </xpath>
         <!-- The toggle mark as done button -->
         <xpath expr="//t[@t-if='isReadonly']" position="after">
             <t t-elif="isToggleMode and currentValue == '01_in_progress'">
-                <button t-if="['kanban', 'list'].includes(this.props.viewType) or this.env.isSmall" class="d-flex align-items-center btn fw-normal p-0" t-on-click.stop="toggleState">
-                    <i t-attf-class="{{ stateIcon(currentValue) }} {{ statusColor(currentValue) }}"></i>
+                <button t-if="['kanban', 'list'].includes(this.props.viewType) or this.env.isSmall" class="d-flex align-items-center btn fw-normal p-0" t-on-click.stop="toggleState" t-att-class="{'o_task_state_list_view': this.props.viewType === 'list'}">
```
Comment (auon-odoo) [inline addons/project/static/src/components/project_task_state_selection/project_task_state_selection.xml:18]: `this.props.viewType === 'list/kanban'` could be a function? `isView('list')`? What do you think?

Comment (auon-odoo) [reply addons/project/static/src/components/project_task_state_selection/project_task_state_selection.xml:18]: Also, doesn't it work without "this"?

File: odoo/addons/base/models/res_config.py [hunk 11]
```diff
@@ -354,8 +354,8 @@ class MyConfigWizard(models.TransientModel):
         *   For a boolean field like 'module_XXX', ``execute`` triggers the immediate
             installation of the module named 'XXX' if the field has value ``True``.
 
-        *   For a selection field like 'module_XXX' composed of 2 string values ('0' and '1'), 
-            ``execute`` triggers the immediate installation of the module named 'XXX' 
+        *   For a selection field like 'module_XXX' composed of 2 string values ('0' and '1'),
+            ``execute`` triggers the immediate installation of the module named 'XXX'
             if the field has the value ``'1'``.
```
Comment (auon-odoo) [inline odoo/addons/base/models/res_config.py:359]: Unnecessary diff :angel:

File: addons/project/views/project_views.xml [hunk 12]
```diff
@@ -1339,24 +1327,30 @@
...
-                                            <field name="partner_id"/>
+                                            <field name="partner_id" class="text-muted" style="display: block; margin-top: 4px;"/>
                                         </t>
+                                        <span t-if="record.allow_milestones.raw_value and record.milestone_id.raw_value" t-attf-class="{{record.has_late_and_unreached_milestone.raw_value ? 'text-danger' : 'text-muted'}}" style="display: block; margin-top: 4px;">
+                                            <field name="milestone_id" options="{'no_open': True}" />
+                                        </span>
                                     </div>
                                 </div>
-                                <div class="o_kanban_record_body">
+                                <div class="o_kanban_record_body" t-attf-class="{{record.is_closed.raw_value ? 'opacity-50' : 'text-muted'}}">
```
Comment (auon-odoo) [inline addons/project/views/project_views.xml:1353]:
```suggestion
                                <div class="o_kanban_record_body" t-att-class="record.is_closed.raw_value ? 'opacity-50' : 'text-muted'">
```
When your whole t-attf is inside curly brackets, you can use t-att and no brackets.
In t-attf, we evaluate what's inside double curly brackets ; in t-att, we evaluate everything.

Comment (auon-odoo) [reply addons/project/views/project_views.xml:1353]: There are other occurrences of this in this PR :slightly_smiling_face:

File: addons/project/static/src/components/subtask_kanban_list/subtask_kanban_list.js [hunk 13]
```diff
@@ -12,6 +12,17 @@ import { standardWidgetProps } from "@web/views/widgets/standard_widget_props";
 export class SubtaskKanbanList extends Component {
     setup() {
         this.actionService = useService("action");
+        this.orm = useService("orm");
+        this.subTaskIds = [];
```
Comment (auon-odoo) [inline addons/project/static/src/components/subtask_kanban_list/subtask_kanban_list.js:16]: is this necessary? :thinking:

**Review by auon-odoo — CHANGES_REQUESTED** [review]
Thanks guys! Sorry I didn't see those at first...

File: addons/project/static/src/components/project_task_state_selection/project_task_state_selection.xml (hunk 10 above)
Comment (bastvdn) [reply addons/project/static/src/components/project_task_state_selection/project_task_state_selection.xml]: Done with the function

File: addons/project/static/src/components/project_task_name_with_subtask_count_char_field/project_task_name_with_subtask_count_char_field.js (hunk 9 above)
Comment (RaoufGhrissi) [reply addons/project/static/src/components/project_task_name_with_subtask_count_char_field/project_task_name_with_subtask_count_char_field.js:9]: i don't know why formatChar, it was there 
why breaking the line ? 
thanks for the info
why ? :p

File: addons/project/static/src/components/project_task_name_with_subtask_count_char_field/project_task_name_with_subtask_count_char_field.js [hunk 14]
```diff
@@ -6,7 +6,7 @@ import { formatChar } from '@web/views/fields/formatters';
 
 class ProjectTaskNameWithSubtaskCountCharField extends CharField {
     get formattedSubtaskCount() {
-        return formatChar(this.props.record.data.allow_subtasks && this.props.record.data.child_text || '');
+        return this.props.record.data.subtask_count ? `(${this.props.record.data.closed_subtask_count}/${this.props.record.data.subtask_count} subtasks)` : '';
```
Comment (auon-odoo) [inline addons/project/static/src/components/project_task_name_with_subtask_count_char_field/project_task_name_with_subtask_count_char_field.js:9]: @RaoufGhrissi Don't wanna break lines? Don't want to use double quotes? :sweat_smile:

File: addons/project/static/src/components/subtask_kanban_list/subtask_kanban_list.js (hunk 13 above)
Comment (auon-odoo) [reply addons/project/static/src/components/subtask_kanban_list/subtask_kanban_list.js:16]: Or is it not? That is the question... :monocle_face:

**Review by auon-odoo — APPROVED** [review]
Still 2 comments left :smile:

File: addons/project/views/project_views.xml [hunk 15]
```diff
@@ -1371,8 +1371,8 @@
                                         <a t-if="!record.project_id.raw_value" class="text-muted" style="font-size: 17px; padding-top: 1.5px;">
                                             <i title="Private Task" class="fa fa-lock"/>
                                         </a>
-                                        <a t-if="record.subtask_count.raw_value and !record.is_private.raw_value">
-                                            <span title="See Subtasks" class="subtask_list_button fa fa-check-square-o me-1"/>
+                                        <a t-if="record.subtask_count.raw_value and !record.is_private.raw_value" class="text-muted">
+                                            <span title="See Subtasks" t-attf-class="{{ record.subtask_count.value &gt; record.closed_subtask_count.value ? subtask_list_button : '' }} fa fa-check-square-o me-1"/>
                                             <t t-out="record.closed_subtask_count.value"/>/<t t-out="record.subtask_count.value"/>
                                         </a>
```
Comment (xavierbol) [inline addons/project/views/project_views.xml]:
```suggestion
                                        <t t-if="!record.is_private.raw_value">
                                            <a t-if="record.subtask_count.value &gt; record.closed_subtask_count.value" class="subtask_list_button text-muted">
                                                <span title="See Subtasks" class="fa fa-check-square-o me-1"/>
                                                <t t-out="record.closed_subtask_count.value"/>/<t t-out="record.subtask_count.value"/>
                                            </a>
                                            <div class="text-muted">
                                                <span t-else="" class="fa fa-check-square-o me-1"/>
                                                <t t-out="record.closed_subtask_count.value"/>/<t t-out="record.subtask_count.value"/>
                                            </div>
                                        </t>
```

Cannot we do something like that? 🤔

File: addons/project/static/src/components/project_task_name_with_subtask_count_char_field/project_task_name_with_subtask_count_char_field.js (hunk 14 above)
Comment (RaoufGhrissi) [reply addons/project/static/src/components/project_task_name_with_subtask_count_char_field/project_task_name_with_subtask_count_char_field.js:9]: haha no @auon-odoo i didn't ignore your comments (we don't do that here), i just pushed yesterday to let xavier see the story of the compiler selector, i will finish it today

File: addons/project/static/src/components/subtask_kanban_list/subtask_kanban_list.js (hunk 13 above)
Comment (RaoufGhrissi) [reply addons/project/static/src/components/subtask_kanban_list/subtask_kanban_list.js:16]: the variable or the initialization ? 
cc @pkyriakou

File: addons/project/static/src/views/project_task_kanban/project_task_kanban_view.scss [hunk 16]
```diff
@@ -2,4 +2,11 @@
     div.o_field_many2many_avatar_user[name="user_ids"] {
         margin-left: 6px;
     }
+
+    .rating_face {
+        font-weight:bold;
+        padding-top:2px;
+        padding-right:3px;
+        font-size:16px;
```
Comment (xavierbol) [inline addons/project/static/src/views/project_task_kanban/project_task_kanban_view.scss:10]:
```suggestion
        font-weight: bold;
        padding-top: 2px;
        padding-right: 3px;
        font-size: 16px;
```

File: odoo/addons/base/models/res_config.py (hunk 11 above)
Comment (xavierbol) [reply odoo/addons/base/models/res_config.py:359]: it is not fixed yet

**Comment by Xavier-Do** [comment]
Upgrade exception [#197](https://runbot.odoo.com/web/#id=197&view_type=form&model=runbot.upgrade.exception) added.

Waiting the forward-port of this pr, the exception will be applied on all builds, and create an inconsistent state for migrations.
**Please forward port this asap up to master without change.** Exception should be forward-ported before the end of the week.

If you need to apply any change before it reaches master, please notify runbot team.

Details:
```
field:project.task.closed_subtask_count
```
cc @KangOl @nseinlet

File: addons/project/views/res_config_settings_views.xml [hunk 17]
```diff
@@ -9,9 +9,6 @@
                 <xpath expr="//form" position="inside">
                     <app data-string="Project" string="Project" name="project" groups="project.group_project_manager">
                         <block title="Tasks Management" id="tasks_management">
-                            <setting help="Split your tasks to organize your work into sub-milestones">
-                                <field name="group_subtask_project"/>
-                            </setting>
```
Comment (xavierbol) [inline addons/project/views/res_config_settings_views.xml:14]: It is better to make it invisible instead of remove it in stable. 😇 
```suggestion
                            <setting invisible="1" help="Split your tasks to organize your work into sub-milestones">
                                <field name="group_subtask_project"/>
                            </setting>
```

It is just to be sure to not break an eventual xpath in a customer db. 🙂

File: addons/project/views/project_views.xml [hunk 18]
```diff
@@ -1181,13 +1177,9 @@
                                     <field name="company_id" groups="base.group_multi_company" optional="hide"/>
                                     <field name="company_id" invisible="1"/>
                                     <field name="activity_ids" string="Next Activity" widget="list_activity" optional="hide"/>
-                                    <field name="date_deadline" attrs="{'invisible': [('is_closed', '=', True)]}" optional="show"/>
+                                    <field name="date_deadline" attrs="{'invisible': [('is_closed', '=', True)]}" optional="hide"/>
                                     <field name="tag_ids" widget="many2many_tags" options="{'color_field': 'color'}" optional="hide"/>
-                                    <field name="rating_last_text" string="Rating" decoration-danger="rating_last_text == 'ko'"
```
Comment (xavierbol) [inline addons/project/views/project_views.xml:1186]: It is better to add `invisible="1"` on `rating_last_text` instead of removing it. 🙂

File: addons/project/views/project_sharing_views.xml [hunk 19]
```diff
@@ -38,9 +38,8 @@
                 <field name="is_closed"/>
                 <field name="displayed_image_id"/>
                 <field name="active"/>
-                <field name="allow_subtasks"/>
-                <field name="child_text"/>
-                <field name="is_closed"/>
```
Comment (xavierbol) [inline addons/project/views/project_sharing_views.xml:43]: Make them invisible instead of removing them. 🙂

File: addons/project/views/project_views.xml [hunk 20]
```diff
@@ -518,11 +518,8 @@
                                 </group>
                             </group>
                             <group>
-                                <group name="group_tasks_managment" string="Tasks Management" col="1" class="row mt16 o_settings_container" groups="project.group_subtask_project,project.group_project_task_dependencies,project.group_project_milestone,project.group_project_recurring_tasks">
+                                <group name="group_tasks_managment" string="Tasks Management" col="1" class="row mt16 o_settings_container" groups="project.group_project_task_dependencies,project.group_project_milestone,project.group_project_recurring_tasks">
                                     <div>
-                                        <setting id="subtask_settings" help="Split your tasks to organize your work into sub-milestones" groups="project.group_subtask_project">
-                                            <field name="allow_subtasks"/>
-                                        </setting>
```
Comment (xavierbol) [inline addons/project/views/project_views.xml:525]: Make the setting element, invisible instead of removing it. 🙂 

Because a custo could do a xpath on it, so it is better to make the element invisible instead of removing it 🙂

File: addons/project/views/project_views.xml [hunk 21]
```diff
@@ -870,16 +867,16 @@
...
                                                 <t t-out="record.milestone_count_reached.value"/>/<t t-out="record.milestone_count.value"/>
                                             </a>
                                         </div>
                                         <field name="activity_ids" widget="kanban_activity"/>
                                     </div>
                                     <div class="oe_kanban_bottom_right">
+                                        <field name="user_id" widget="many2one_avatar_user" domain="[('share', '=', False)]" style="padding-right: 4px;"/>
                                         <field t-if="record.last_update_status.value &amp;&amp; widget.editable" name="last_update_status" widget="project_state_selection"/>
                                         <span t-if="record.last_update_status.value &amp;&amp; !widget.editable" t-att-class="'o_status_bubble mx-0 o_color_bubble_' + record.last_update_color.value" t-att-title="record.last_update_status.value"></span>
-                                        <field name="user_id" widget="many2one_avatar_user" domain="[('share', '=', False)]"/>
```
Comment (xavierbol) [inline addons/project/views/project_views.xml:882]: Make it invisible. 🙂

File: addons/project/views/project_views.xml [hunk 22]
```diff
@@ -1024,7 +1021,6 @@
             <field eval="2" name="priority"/>
             <field name="arch" type="xml">
                 <form string="Task" class="o_form_project_tasks" js_class="project_task_form">
-                    <field name="allow_subtasks" invisible="1" />
```
Comment (xavierbol) [inline addons/project/views/project_views.xml:1027]: We could keep it since it is invisible. 🙂

File: addons/project/views/project_views.xml [hunk 23]
```diff
@@ -1199,10 +1191,10 @@
                                     <field name="allow_milestones" invisible="1"/>
                                     <field name="parent_id" invisible="1" />
                                     <field name="is_closed" invisible="1" />
-                                    <field name="priority" widget="priority" optional="show" nolabel="1"/>
-                                    <field name="state" widget="project_task_state_selection" optional="show" nolabel="1"/>
-                                    <field name="child_text" invisible="1"/>
-                                    <field name="allow_subtasks" invisible="1"/>
```
Comment (xavierbol) [inline addons/project/views/project_views.xml:1205]: We should keep them invisible on the view to avoid breaking custo code, in the case they did a xpath on those fields. 🙂

File: addons/project/views/project_views.xml [hunk 24]
```diff
@@ -1212,26 +1204,22 @@
...
+                                    <field name="parent_id" optional="hide" groups="base.group_no_one"/>
                                     <field name="user_ids" widget="many2many_avatar_user" optional="show" domain="[('share', '=', False), ('active', '=', True)]"/>
                                     <field name="company_id" optional="hide" groups="base.group_multi_company" />
                                     <field name="company_id" invisible="1"/>
                                     <field name="activity_ids" string="Next Activity" widget="list_activity" optional="hide"/>
-                                    <field name="date_deadline" attrs="{'invisible': [('is_closed', '=', True)]}" optional="show" />
+                                    <field name="date_deadline" attrs="{'invisible': [('is_closed', '=', True)]}" optional="hide" />
                                     <field name="tag_ids" widget="many2many_tags" options="{'color_field': 'color'}" optional="hide"/>
-                                    <field name="rating_last_text" string="Rating" decoration-danger="rating_last_text == 'ko'"
```
Comment (xavierbol) [inline addons/project/views/project_views.xml:1222]: Make it invisible 🙂

File: addons/project/views/project_views.xml [hunk 25]
```diff
@@ -1317,8 +1305,6 @@
                     <field name="activity_state"/>
                     <field name="rating_count"/>
                     <field name="rating_avg"/>
-                    <field name="allow_subtasks"/>
-                    <field name="child_text"/>
```
Comment (xavierbol) [inline addons/project/views/project_views.xml:1321]: Make them invisible instead. 🙂

File: addons/project/views/project_views.xml [hunk 26]
```diff
@@ -1402,11 +1414,11 @@
                     <field name="is_private" invisible="1" />
                     <field name="sequence" invisible="1" readonly="1"/>
                     <field name="allow_milestones" invisible="1"/>
+                    <field name="subtask_count" invisible="1"/>
+                    <field name="closed_subtask_count" invisible="1"/>
                     <field name="id" optional="hide"/>
-                    <field name="priority" widget="priority" optional="show" nolabel="1"/>
-                    <field name="state" widget="project_task_state_selection" optional="show" nolabel="1"  options="{'hide_label': 1, 'is_toggle_mode': false}"/>
-                    <field name="child_text" invisible="1"/>
-                    <field name="allow_subtasks" invisible="1"/>
```
Comment (xavierbol) [inline addons/project/views/project_views.xml:1409]: Make them invisible instead. 🙂

File: addons/project/static/src/views/project_task_kanban/project_task_kanban_view.scss (hunk 16 above)
Comment (bastvdn) [reply addons/project/static/src/views/project_task_kanban/project_task_kanban_view.scss]: Done

File: addons/project/views/res_config_settings_views.xml (hunk 17 above)
Comment (bastvdn) [reply addons/project/views/res_config_settings_views.xml]: Done

File: addons/project/views/project_views.xml (hunk 18 above)
Comment (bastvdn) [reply addons/project/views/project_views.xml:1186]: Done

File: addons/project/views/project_sharing_views.xml (hunk 19 above)
Comment (bastvdn) [reply addons/project/views/project_sharing_views.xml]: Done

File: addons/project/views/project_views.xml (hunk 20 above)
Comment (bastvdn) [reply addons/project/views/project_views.xml]: Done

File: addons/project/views/project_views.xml (hunk 21 above)
Comment (bastvdn) [reply addons/project/views/project_views.xml:882]: Done

File: addons/project/views/project_views.xml (hunk 22 above)
Comment (bastvdn) [reply addons/project/views/project_views.xml]: Done

File: addons/project/views/project_views.xml (hunk 23 above)
Comment (bastvdn) [reply addons/project/views/project_views.xml]: Done

File: addons/project/views/project_views.xml (hunk 24 above)
Comment (bastvdn) [reply addons/project/views/project_views.xml:1222]: done

File: addons/project/views/project_views.xml (hunk 25 above)
Comment (bastvdn) [reply addons/project/views/project_views.xml]: Done

File: addons/project/views/project_views.xml (hunk 26 above)
Comment (bastvdn) [reply addons/project/views/project_views.xml]: Done

**Comment by bastvdn** [comment]
@xavierbol Thanks for the review, tested the views, everything's seems good :pray:

File: addons/project/static/src/components/subtask_kanban_list/subtask_kanban_list.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (xavierbol) [reply addons/project/static/src/components/subtask_kanban_list/subtask_kanban_list.js:24]: We will discuss about it, but I am not sure if it is a good idea to remove the is_closed, or maybe we will have to index=True the state field? 

cf8bb93

File: addons/project/views/project_views.xml (hunk 18 above)
Comment (xavierbol) [reply addons/project/views/project_views.xml:1186]: Nope :)

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work! 😁 🥳

**Comment by xavierbol** [comment]
robodoo r+
